primary/src/metrics.rs

```diff
@@ -66,6 +108,9 @@ pub struct PrimaryMetrics {
     pub parent_requests_header_waiter: IntGaugeVec,
     /// Number of elements in pending list of certificate_waiter
     pub pending_elements_certificate_waiter: IntGaugeVec,
+
+    // Core metrics sneaking in
+    pub core_metrics: Arc<CoreCounters>,
```

Comment:
haha really sneaking in! Yeah I would suggest keeping those out of the `PrimaryMetrics` and generally outside of the concept of metrics that we use for monitoring purposes. I would suggest to initialise and inject them independently of the above struct.